plugins/TPChannelFilter.cpp

```diff
@@ -136,14 +151,33 @@ TPChannelFilter::do_work()
 
     // Actually do the removal for payload TPSets. Leave heartbeat TPSets unmolested
     if (tpset->type == TPSet::kPayload) {
+
+      if (m_first_tp) {
+        if (m_use_latency_monit && m_use_latency_offset) {
+          m_initial_offset = (duration_cast<milliseconds>(system_clock::now().time_since_epoch()).count()) -
+                             (tpset->objects.begin()->time_start * m_clock_ticks_to_ms);
+        }
+        m_first_tp = false;
+      }
+
+      if (m_use_latency_monit) {
+        // block to report latency to opmon
+        m_system_time =
+          std::chrono::duration_cast<std::chrono::milliseconds>(system_clock::now().time_since_epoch()).count();
+        m_data_vs_system_time.store(fabs(m_system_time - (tpset->start_time * m_clock_ticks_to_ms) - m_initial_offset));
+        TLOG_DEBUG(TLVL_DEBUG_ALL) << "[TPCF] "
+                                   << fabs(
+                                        (m_system_time - (tpset->start_time * m_clock_ticks_to_ms) - m_initial_offset));
```

Comment:
```suggestion
        TLOG_DEBUG(TLVL_DEBUG_ALL) << "[TPCF] "
                                   << fabs((m_system_time - (tpset->start_time * m_clock_ticks_to_ms) - m_initial_offset));
```
That clang formatting script isn't that great...